### PR TITLE
Adds Orange + Purple Bandanas and Bandana Colouring, Taj/Una Bandana Sprite Tweaks + Mask Adjustment Refactor

### PR DESCRIPTION
**Pull request by KasparoVy**
:cl:
rscadd: Orange and purple bandanas for all station species.
rscadd: The ability to colour bandanas with a crayon in a washing machine.
tweak: Refactors the bandana adjustment system.
tweak: Minor adjustments to Tajaran and Unathi bandana east/west sprites.
tweak: Adjusting a bandana will now take if off of your head/face and put it in an available hand. If both hands are available, it goes in the selected hand.
fix: Adjusting a coloured bandana will no longer revert it to the original coloured icons anymore.
fix: Adjusting a bandana from mask-style to hat-style means the bandana won't obscure your identity anymore.
/:cl:

I know purple and orange seem like pretty random colours, but they were the only ones there weren't existing bandana sprites for.
Also refactors bandana adjustment. I made the change because now it won't revert back to the original icon of the bandana you coloured after you adjust it twice after colouring it with a crayon in the washing machine. Additionally fixes a bug where adjusting a bandana kept it in the same slot regardless of whether it was meant to be worn as a hat or a mask, and another where adjusting a bandana didn't change whether or not it obscured your identity, even when you were wearing it as a hat. Thanks @FalseIncarnate for bringing that up

**Tested in the following ways:**
- Multiple washes, multiple recolourings: Description stayed 'The colours are a bit dodgy' and the  name was appropriate for the colour. No sprite issues.
- The above, but adjusting the mask between washes. No issues discovered.
- Spammed the adjustment, changing from the head to the face slot at random intervals: Same as above, no sprite issues, description was as intended, name of the item was as intended.
- Combination of the two: No issues either.
- Adjusting the bandana to hat-style from mask-style with ID off: Bandana no longer obscured identity.
- Adjusting the bandana to mask-style from hat-style with ID off: Bandana now obscures identity.

**ADJUSTING BANDANA**
- Adjusting bandana while on head with both hands full: Bandana got adjusted dropped on ground.
- Adjusting bandana while on head with only the right hand empty and the left hand selected: Bandana got adjusted and put in the right hand.
- Adjusting bandana while on head with only the right hand empty and the right hand selected: Bandana got adjusted and put in the right hand.
- Adjusting bandana while on head with only the left hand empty and the left hand selected: Bandana got adjusted and put in the left hand.
- Adjusting bandana while on head with only the left hand empty and the right hand selected: Bandana got adjusted and put in the left hand.
- Adjusting bandana while on head with both hands empty and the right hand selected: Bandana got adjusted and put in the right hand.
- Adjusting bandana while on head with both hands empty and the leftt hand selected: Bandana got adjusted and put in the left hand.
- _DOING THE EXACT SAME TESTS EXCEPT WITH THE BANDANA ON THE FACE_: Everything worked the same way, as intended.

The tweak I made to the East and West directions of the Tajara and Unathi bandana sprites were just to make it fit the muzzle better and make it look like the cloth was pulled taut over the mouth as was the case with the Vulpkanin bandana (and the Vox) to a degree. Slightly less ghetto looking, fits to the muzzle a bit better in the same fashion as the Vulpkanin bandanas

### CODE REVIEW

File: code/game/machinery/washing_machine.dm [hunk 1]
```diff
@@ -154,6 +168,14 @@
 					S.item_color = wash_color
 					S.name = new_shoe_name
 					S.desc = new_desc
+			if(new_bandana_icon_state && new_bandana_name)
+				for(var/obj/item/clothing/mask/bandana/M in contents)
+					//world << "DEBUG: YUP! FOUND IT!"
```
Comment (Markolie) [inline code/game/machinery/washing_machine.dm]: Best to just get rid of this.

File: code/modules/clothing/clothing.dm [hunk 2]
```diff
@@ -289,8 +289,7 @@ BLIND     // can't see anything
 		if(!user.canmove || user.stat || user.restrained())
 			return
 		if(src.mask_adjusted == 1)
-			src.icon_state = initial(icon_state)
-			gas_transfer_coefficient = initial(gas_transfer_coefficient)
```
Comment (Markolie) [inline code/modules/clothing/clothing.dm]: Are you sure this should be deleted?

Comment (KasparoVy) [inline code/modules/clothing/clothing.dm]: Not at all, that was a mistake. Back it goes!

File: code/game/machinery/washing_machine.dm (hunk 1 above)
Comment (KasparoVy) [inline code/game/machinery/washing_machine.dm]: Roger roger.

**Comment by FalseIncarnate** [comment]
Would it be possible to include a small change to make it so that bandanas don't hide your identity unless they are actually on your face? It seems silly that when worn as a hat, they somehow obscure your face despite only visibly hiding your hair.

**Comment by KasparoVy** [comment]
That's an issue with adjustable masks in general, as it tends to happen with the balaclava, too, I believe
I'll see about fixing that

**Comment by Fox-McCloud** [comment]
@FalseIncarnate 

Another problem: we're pretty much out of inventory flags and need to split them, but not on really wants to do it.

TG did it, but we never got around to.

**Comment by FalseIncarnate** [comment]
I believe that would definitely be beyond the scope or intent of this PR fox, but good to know.

As for the flags on the masks / bandanas, it simply is a case of toggling the HIDES_FACE bitflag when you switch between mask and hat mode.

**Comment by Fox-McCloud** [comment]
@FalseIncarnate 

Oh, no no, I wasn't suggesting it be done, I was just informing you of why it's so convoluted and a mess.

**Comment by KasparoVy** [comment]
@FalseIncarnate The thing is that, when you turn the bandana into a form that can be worn hat-style, it remains on your face, too-- while if you unequip the bandana and wear it again, it goes on your head. That's still a bug as well, I fixed the issue you're talking about but the one I found's proving to be a little finicky

**Comment by KasparoVy** [comment]
I was out most of the day, but I got a fix for those bugs coming in a minute. Just testing again
Sorry for the delay

**Comment by KasparoVy** [comment]
That's all I've got unless there are any other suggestions/critiques. I'll be able to address them after midnight GMT until 8AM GMT on 2016/10/2 and the following days (up to and including Saturday) in the same time span

**Pull request merged**